sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -277,6 +283,25 @@ def test_imputation_pipeline_grid_search():
     gs.fit(X, Y)
 
 
+def test_mice_pipeline_grid_search():
+    # Test imputation within a pipeline + gridsearch.
+    pipeline = Pipeline([('imputer', MICEImputer(missing_values=0,
+                                                 n_imputations=1,
+                                                 n_burn_in=1)),
+                         ('tree', tree.DecisionTreeRegressor(random_state=0))])
+
+    parameters = {
+        'imputer__initial_strategy': ["mean", "median", "most_frequent"]
+    }
+
+    n = 100
+    d = 10
+    X = sparse_random_matrix(n, d, density=0.50).toarray()
+    Y = np.random.random((n, d))
+    gs = GridSearchCV(pipeline, parameters)
+    gs.fit(X, Y)
+
+
 def test_imputation_pickle():
```

Comment:
Will do. It only seems to pass the pickling test if `random_state` is set. Is that something to be worried about? The algorithm does have a stochastic step (sampling form estimator posterior).

Reply:
This makes sense, since the correctness of the pickling is tested on the basis of prediction/transformation being the same. But isn't pickling tested in common tests?

Alternatively, we could store the random state on the object so that transform is deterministic in all cases. Please take a look at other estimators with stochastic prediction, perhaps indicated by presence of a `random_state_` attribute.

Reply:
Thanks. I'm going to copy `kmeans` and do this in `fit_transform`:

```
        self.random_state_ = getattr(self, "random_state_", check_random_state(self.random_state))
```

Reply:
Feel free to ignore my earlier questions about this issue - I believe this covers it.